src/pages/MapPage/components/SearchBar/SearchBar.js

```diff
@@ -42,35 +44,111 @@ const useStyles = makeStyles((theme) => ({
   }
 }))
 const SearchBar = React.forwardRef((props, ref) => {
-  const { menuControls, ...otherProps } = props
+  const {
+    menuControls,
+    setPlacePosition,
+    search,
+    setSearch,
+    setPlaceName,
+    ...otherProps
+  } = props
   const classes = useStyles()
   const menuControl = useMenu()
   const [open, changeOpen] = useState(false)
   const toggle = () => changeOpen(!open)
   const { enqueueSnackbar } = useSnackbar()
-  const { currentStep } = useMissionValue()
+  const { currentStep, setMapCenter } = useMissionValue()
+  const bounds = {
+    north: 24.791768,
+    south: 24.782687,
+    east: 121.000443,
+    west: 120.995352
+  }
+  const [positionName, setPositionName] = useState('')
+  useEffect(() => {
+    setPlaceName(positionName)
+    document.getElementById('inputBase').value = positionName
+  }, [positionName, setPlaceName])
+  const { ref: materialRef } = usePlacesWidget({
+    onPlaceSelected: (Place) => {
+      if (Place.place_id !== undefined) {
+        setSearch(true)
+        const placeName = document
+          .getElementById('inputBase')
+          .value.split('新竹市東區')[1]
+        if (placeName === undefined) {
+          enqueueSnackbar('地址超過搜尋範圍', { varient: 'error' })
+        } else {
+          if (placeName.split('路')[1] === undefined) {
+            setPositionName(placeName)
+          } else {
+            setPositionName(placeName.split('路')[1])
+          }
+          setPlacePosition(Place.geometry.location)
+          setMapCenter(Place.geometry.location)
+        }
```

Comment:
Maybe add some comment here about what it's used to?

Reply:
Okay already add !